### PR TITLE
[READY] Restricts the cook's CQC to the kitchen and kitchen backroom

### PR DESCRIPTION
**Pull request by nervere**
:cl: Nervere and subject217
balance: The cook's CQC now only works when in the kitchen or the kitchen backroom.
/:cl:

[why]: The reason why the cook got CQC, other than it being a meme, was that people would often break into the kitchen and the cook would have a hard time defending himself. There's really no problem with this, except when the cook can also use CQC in the bar, which tends to be a pretty big area that the cook doesn't even need to defend. The bartender has a shotgun for the purpose of defending his bar, after all. 

I removed CQC working in the cafeteria for the same reason, but `/area/crew_quarters/cafeteria` isn't even used as an area in any of the maps so it doesn't even have an effect. Kitchen backroom isn't used as an area, either, but @81Denton plans to change that soon, so I've added it to the list of areas where CQC is able to be used.

### CODE REVIEW

File: code/datums/martial/cqc.dm [hunk 1]
```diff
@@ -10,8 +10,7 @@
 	block_chance = 75
 	var/just_a_cook = FALSE
 	var/static/list/areas_under_siege = typecacheof(list(/area/crew_quarters/kitchen,
-														/area/crew_quarters/cafeteria,
-														/area/crew_quarters/bar))
+														/area/crew_quarters/kitchen/backroom))
```
Comment (ShizCalev) [inline code/datums/martial/cqc.dm]: this is a typecache, `/area/crew_quarters/kitchen/backroom` would be included under `/area/crew_quarters/kitchen`

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by ATH1909** [comment]
So now chefs are going to have to clear out their kitchens (or get the CE's blueprints, depending on how their interaction with "x effect is only active in y area" types of things works) in order to make rage cages for them to use CQC in?

**Comment by subject217** [comment]
people literally playing chef for the instant stun grabs and 2 hit knockouts isn't really what the role was ever for, this is 100% necessary tbh

**Comment by BeeSting12** [comment]
chefs getting cqc is dumb anyway lol

**Comment by optimumtact** [comment]
@ShizCalev it's not exactly clear to me what you're trying to request in terms of changes

**Comment by ShizCalev** [comment]
@optimumtact it's a typecache. it's like writing `istype(yourmom, /area) || istype(yourmom, /area/crew_quarters/kitchen)`. it's redundant to have both paths present in it as the prior already covers it the latter.

It can either be removed, or this can be simplified further by just removing the typecache and replacing the is_type_in_typecache check @ l25 to `istype(A, /area/crew_quarters/kitchen)` instead.

**Pull request merged**